### PR TITLE
Refactor API structure 🔭

### PR DESCRIPTION
**Pull request by laurmaedje**
- RawBibliography / Bibliography
- Formatting functions moved to ChunkExt
- Private RawChunk and public Chunk
- RawChunk::Resolve -> RawChunk::Abbreviation
- Improved date representation
- More public functions made public
- Take &[Chunk] in all constructors

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,13 +1,217 @@
...
+        self.get(key)
+            .ok_or_else(|| anyhow!("The {} field is not present", key))
+            .and_then(|chunks| chunks.parse::<T>())
+    }
+}
+
+macro_rules! fields {
+    ($($getter:ident: $field_name:expr $(=> $res:ty)?),* $(,)*) => {
+        $(
+            #[doc = "Get and parse the `"]
```
Comment (reknih) [inline src/lib.rs:102]: I like the automatically generated docs

File: src/lib.rs [hunk 2]
```diff
@@ -1,13 +1,217 @@
...
+    fn parse<T: Type>(&self) -> anyhow::Result<T>;
+
+    /// Format the chunks in sentence case.
+    fn format_sentence(&self) -> String;
+
+    /// Format the chunks verbatim.
+    fn format_verbatim(&self) -> String;
+}
+
+impl ChunksExt for [Chunk] {
```
Comment (reknih) [inline src/lib.rs:169]: Also nice having those as traits

File: src/types.rs [hunk 3]
```diff
@@ -270,28 +270,37 @@ impl fmt::Display for Person {
...
     /// Indicates whether the sources are sure about the date.
     pub is_uncertain: bool,
     /// Indicates the specificity of the date value.
     pub is_approximate: bool,
-    /// The date's value, or its start point if `range_end.is_some()`.
-    pub value: DateKind,
-    /// If this is Some, the date is a range (`Date.value .. Date.range_end`).
-    pub range_end: Option<DateKind>,
+    /// The date or the date range.
+    pub value: DateValue,
```
Comment (reknih) [inline src/types.rs:281]: I see that this is now enum-tastic

File: src/types.rs [hunk 4]
```diff
@@ -537,7 +543,7 @@ impl Date {
 }
 
 /// Used to resolve month abbreviations to their respective values.
-pub fn get_month_for_abbr(month: &str) -> Option<(String, u8)> {
+pub(crate) fn get_month_for_abbr(month: &str) -> Option<(String, u8)> {
```
Comment (reknih) [inline src/types.rs:546]: Never seen that before

**Review by reknih — APPROVED** [review]

File: src/types.rs (hunk 3 above)
Comment (laurmaedje) [reply src/types.rs:281]: Indeed!

File: src/types.rs (hunk 4 above)
Comment (laurmaedje) [reply src/types.rs:546]: There's also `pub(in <module_name_here>)`, but I've never had a need for that.

**Pull request merged**